templates/experiments/single_experiment_home.html

```diff
@@ -212,54 +212,244 @@ <h3 class="font-bold text-lg">Parent Experiments:
-              >
-                <div class="flex gap-2">
-                  <div class="w-72 my-2">
-                    <input x-model="participants" name="participants" id="participants-filter" placeholder="Filter participants...">
+              <div x-data="filterComponent" @click.outside="filterData.showFilters = false">
+              <!-- Filter Button -->
+                <div class="flex items-center">
+                  <button @click="toggleFilters()" class="btn btn-sm btn-outline">
+                    <i class="fa-solid fa-filter"></i> Filter <span x-show="getActiveFilterCount() > 0" x-cloak>(<span x-text="getActiveFilterCount()"></span>)</span>
+                  </button>
+
+                  <div x-show="filterData.loading" x-cloak class="ml-2">
+                    <i class="fa-solid fa-spinner fa-spin"></i>
                   </div>
-                  {% if perms.annotations.view_customtaggeditem %}
-                    <div class="w-72 my-2">
-                      <select x-model="tags" id="tag-multiselect-filter" name="tags" multiple
-                              placeholder="Filter tags..."
-                              autocomplete="off">
-                        <option value="">Filter tags...</option>
-                        {% for tag in available_tags %}
-                          <option value="{{ tag.name }}">{{ tag.name }}</option>
-                        {% endfor %}
-                      </select>
-                    </div>
-                  {% endif %}
-                  <div class="form-control">
-                    <label class="label cursor-pointer">
-                      <span class="label-text mr-2">Include API Sessions</span>
-                      <input type="checkbox" name="show-all" class="toggle" />
-                    </label>
+
+                  <!-- Clear filters button -->
+                  <button
+                    x-show="getActiveFilterCount() > 0 && !filterData.loading"
+                    @click="clearAllFilters()"
+                    class="btn btn-square btn-sm btn-ghost"
+                    title="Clear all filters" x-cloak>
+                    x
+                  </button>
+                </div>
+
+              <!-- Filters Panel -->
+                <div
+                  x-show="filterData.showFilters"
+                  class="absolute left-0 mt-2 p-4 bg-white border border-gray-200 rounded-lg shadow-lg z-10 w-fit"
+                >
+                  <div class="space-y-3">
+                    <template x-for="(filter, index) in filterData.filters" :key="index">
+                      <div class="flex items-center gap-3">
+                      <!-- Where/AND Label -->
+                        <div class="text-gray-500 font-medium w-16">
+                          <span x-text="index === 0 ? 'Where' : 'AND'"></span>
+                        </div>
+                      <!-- Column Selector -->
+                        <div class="relative py-2">
```

Comment:
spacing is handled by `gap-2` and `space-y-2`
```suggestion
                        <div class="relative">
```